src/frontend/src/lib/utils/findWebAuthnFlows.ts

```diff
@@ -35,32 +43,58 @@ export const findWebAuthnFlows = ({
   currentOrigin,
   relatedOrigins,
 }: Parameters): WebAuthnFlow[] => {
+  // We need the helpers inside so that when `canisterConfig` is accessed, it already exists.
+  // The devices are expected to be ordered by recently used already
+  // Move devices registered on the new flow origins to the end using toSorted (preserving relative order within groups)
+  const newFlowOrigins = canisterConfig.new_flow_origins[0] ?? [];
+  const isInNewFlow = (credentialData: CredentialData): boolean => {
+    const origin = credentialData.origin ?? II_LEGACY_ORIGIN;
+    return newFlowOrigins.some((o) => isSameOrigin(o, origin));
+  };
+  const sortNewFlowOriginsToEnd = (
+    a: CredentialData,
+    b: CredentialData,
+  ): number => {
+    const aIn = isInNewFlow(a);
+    const bIn = isInNewFlow(b);
+    // Keep the order if both are in the new flow or both are not
+    if (aIn === bIn) return 0;
+    // Move the one that is in the new flow to the end
+    return aIn ? 1 : -1;
+  };
+
   const currentRpId = new URL(currentOrigin).hostname;
   const relatedRpIds = relatedOrigins.map(
     (relatedOrigin) => new URL(relatedOrigin).hostname,
   );
 
-  // The devices are expected to be ordered by recently used already
-  const orderedDeviceRpIds = [
-    ...new Set(
-      devices
-        // Device origin to RP ID (hostname)
-        .map((device) =>
-          device.origin === currentOrigin ||
-          (currentOrigin === II_LEGACY_ORIGIN && isNullish(device.origin))
-            ? undefined
-            : new URL(device.origin ?? II_LEGACY_ORIGIN).hostname,
-        )
-        // Filter out RP IDs that are not within `relatedRpIds`
-        .filter((rpId) => isNullish(rpId) || relatedRpIds.includes(rpId)),
-    ),
-  ];
-
-  // Create steps from `deviceRpIds`, currently that's one step per RP ID
-  const steps: WebAuthnFlow[] = orderedDeviceRpIds.map((rpId) => ({
-    rpId,
-    useIframe: nonNullish(rpId) && rpId !== currentRpId,
-  }));
+  // Sort devices in place
+  devices.sort(sortNewFlowOriginsToEnd);
+  // Create steps from `devices`, currently that's one step per RP ID
+  const steps: WebAuthnFlow[] = devices
+    // Device origin to RP ID (hostname)
+    .map((device: CredentialData) =>
+      device.origin === currentOrigin ||
+      (currentOrigin === II_LEGACY_ORIGIN && isNullish(device.origin))
+        ? undefined
+        : new URL(device.origin ?? II_LEGACY_ORIGIN).hostname,
+    )
+    // Filter out RP IDs that are not within `relatedRpIds`
+    .filter(
+      (rpId: string | undefined) =>
+        isNullish(rpId) || relatedRpIds.includes(rpId),
+    )
+    // Remove duplicates
+    .reduce((rpIds: Array<string | undefined>, rpId: string | undefined) => {
+      if (rpIds.includes(rpId)) {
+        return rpIds;
+      }
+      return [...rpIds, rpId];
+    }, [])
+    .map((rpId) => ({
+      rpId,
+      useIframe: nonNullish(rpId) && rpId !== currentRpId,
+    }));
```

Comment:
```suggestion
const steps: WebAuthnFlow[] = [
  ...new Set(
    devices
      .map(d =>
        d.origin === currentOrigin || (currentOrigin === II_LEGACY_ORIGIN && isNullish(d.origin))
          ? undefined
          : new URL(d.origin ?? II_LEGACY_ORIGIN).hostname,
      )
      .filter(rpId => isNullish(rpId) || relatedRpIds.includes(rpId)),
  ),
].map(rpId => ({
  rpId,
  useIframe: nonNullish(rpId) && rpId !== currentRpId,
}));
```

Why not use a Set here to remove duplicates instead of reduce? 
Set has better performance, and automatically infers removing duplicates without the need to understand the logic of the reduce

Reply:
Let's keep the code, as the added complexity serves a purpose.